test/unit/project_type/test_git.py

```diff
@@ -178,3 +178,15 @@ def expect_side_effect(patterns, *args, **kwargs):
         git_executor._execute_git_remote_command('fake command', cwd=None, timeout=0, log_msg_queue=MagicMock())
 
         self.assertTrue(matched_prompt, "The password prompt was not matched by pexpect")
+
+    def test_execute_raises_broken_pipe_error_on_last_try(self):
+        git_executor = _GitRemoteCommandExecutor()
+        self.patch('app.project_type.git.sleep')
+        mock_manager = self.patch('app.project_type.git.Manager')
+
+        def manager_init_side_effect(*args):
+            raise BrokenPipeError('')
```

Comment:
This works in console:
```python
from unittest.mock import MagicMock
m = MagicMock()
m.test.side_effect = BrokenPipeError
m.test()
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python3/3.4.2/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/mock.py", line 896, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.4.2/Frameworks/Python.framework/Versions/3.4/lib/python3.4/unittest/mock.py", line 952, in _mock_call
    raise effect
BrokenPipeError
```

We also do exception side effects in test_project_type.py and test_autoversioning.py:
```python
self.check_output_mock.side_effect = [subprocess.CalledProcessError(1, 'fake')] 
```